pkg/spdk/disk/types.go

```diff
@@ -0,0 +1,139 @@
+package disk
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"slices"
+
+	"github.com/pkg/errors"
+
+	commonTypes "github.com/longhorn/go-common-libs/types"
+	spdksetup "github.com/longhorn/go-spdk-helper/pkg/spdk/setup"
+	helperutil "github.com/longhorn/go-spdk-helper/pkg/util"
+
+	"github.com/longhorn/longhorn-spdk-engine/pkg/util"
+)
+
+type BlockDiskDriver string
+
+const (
+	BlockDiskDriverAuto       = BlockDiskDriver("auto")
+	BlockDiskDriverAio        = BlockDiskDriver("aio")
+	BlockDiskDriverNvme       = BlockDiskDriver("nvme")
+	BlockDiskDriverVirtioPci  = BlockDiskDriver("virtio-pci")
+	BlockDiskDriverVirtioScsi = BlockDiskDriver("virtio-scsi")
+	BlockDiskDriverVirtioBlk  = BlockDiskDriver("virtio-blk")
+)
+
+type BlockDiskSubsystem string
+
+const (
+	BlockDiskSubsystemVirtio = BlockDiskSubsystem("virtio")
+	BlockDiskSubsystemPci    = BlockDiskSubsystem("pci")
+	BlockDiskSubsystemNvme   = BlockDiskSubsystem("nvme")
+	BlockDiskSubsystemScsi   = BlockDiskSubsystem("scsi")
+)
+
+const (
+	UioDriverUioPciGeneric = "uio_pci_generic"
+)
+
+type BlockDiskType string
+
+const (
+	BlockDiskTypeDisk = BlockDiskType("disk")
+	BlockDiskTypeLoop = BlockDiskType("loop")
+)
+
+type DiskStatus struct {
+	Bdf          string
+	Type         string
+	Driver       string
+	Vendor       string
+	Numa         string
+	Device       string
+	BlockDevices string
+}
+
+func GetDiskDriver(diskDriver, diskPathOrBdf string) (BlockDiskDriver, error) {
+	if isBDF(diskPathOrBdf) {
+		return getDiskDriverForBDF(diskDriver, diskPathOrBdf)
+	}
+
+	return getDiskDriverForPath(diskDriver, diskPathOrBdf)
+}
+
+func getDiskDriverForBDF(diskDriver, bdf string) (BlockDiskDriver, error) {
+	executor, err := helperutil.NewExecutor(commonTypes.ProcDirectory)
```

Comment:
NIT: Use the executor in `go-common-lib` instead.

Reply:
It is a wrapper of `NewNamespaceExecutor` in go-common-lib.